proto-lens/src/Data/ProtoLens/Message.hs

```diff
@@ -231,3 +260,17 @@ reverseRepeatedFields fields x0
     reverseListField x (FieldDescriptor _ _ (RepeatedField _ f))
         = over f reverse x
     reverseListField x _ = x
+
+newtype Registry = Registry (Map.Map T.Text SomeMessageType)
```

Comment:
Let's add Haddocks for these functions, maybe:

    -- | A set of known message types.  Can help encode/decode protobufs containing
    -- @Data.ProtoLens.Any@ values in a more human-readable text format.
    -- 
    -- Registries can be combined using their 'Monoid' instance. 
    newtype Registry = ...

    -- | Build a 'Registry' containing a single proto type.
    register :: ...

    -- | Look up a message type by name (e.g., @"google.protobuf.FloatValue"@).
    -- The URL corresponds to the field @google.protobuf.Any.type_url@.
    lookupRegistered :: ...